test/SlowTests/Server/Documents/OngoingTasks/PinOnGoingTaskToMentorNode.cs

```diff
@@ -232,15 +232,12 @@ public async Task Can_Fail_Over_When_Removing_Mentor_Node_Etl()
             
             Assert.True(WaitForDocument<User>(dest, "users/1", u => u.Name == "Joe Doe", 30_000));
             
-            await ActionWithLeader(l =>
-            {
-                l.ServerStore.RemoveFromClusterAsync(responsibleNodeNodeTag);
-            });
+            await ActionWithLeader(l => l.ServerStore.RemoveFromClusterAsync(responsibleNodeNodeTag));
             
             using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
             {
-                var waitForNotPassive = await mentorNode.ServerStore.Engine.WaitForLeaveState(RachisState.Passive, cts.Token);
-                Assert.True(waitForNotPassive);
+                var waitForPassive = await mentorNode.ServerStore.Engine.WaitForState(RachisState.Passive, cts.Token);
```

Comment:
this is not the same thing? it was 'WaitForLeaveState' and now you have 'WaitForState' ?

Reply:
yes, it was wrong and it worked because of the code `ActionWithLeader`
it got action which create a task that wasn't awaited, so basically `RemoveFromClusterAsync` wasn't executed run at this point